castle/test/commands/authenticate_test.py

```diff
@@ -21,6 +21,18 @@ def test_build(self):
         self.assertEqual(command.endpoint, 'authenticate')
         self.assertEqual(command.data, payload)
 
+    def test_build_no_event(self):
+        payload = {'user_id': '1234'}
+
+        with self.assertRaises(InvalidParametersError):
+            CommandsAuthenticate({}).build(payload)
+
+    def test_build_no_user_id(self):
+        payload = {'event': '$login.authenticate'}
```

Comment:
Let me know if I'm missing something here. But events such as `$login.failed` does not require a user_id to be tracked to Castle. Ie. it shouldn't trigger an error is user_id is not set

Reply:
If so, then we need to remove it in both `castle-ruby` and `castle-python`.

Reply:
nvm. I was confused. No use in authenticating $login.failed events.